src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStoragePublisher.java

```diff
@@ -265,63 +271,69 @@ public StorageAccountInfo getStorageAccount() {
 		return storageAcc;
 	}
 
-	public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
-			BuildListener listener) throws InterruptedException, IOException {
+	public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath ws, @Nonnull Launcher launcher, @Nonnull TaskListener listener) throws InterruptedException, IOException {
 
 		// Get storage account and set formatted blob endpoint url.
 		StorageAccountInfo strAcc = getStorageAccount();
+		final EnvVars envVars = run.getEnvironment(listener);
 
 		// Resolve container name
-		String expContainerName = Utils.replaceTokens(build, listener,
-				containerName);
-		if (expContainerName != null) {
+		 String expContainerName = Util.replaceMacro(containerName, envVars);
+        if (!Utils.isNullOrEmpty(expContainerName)) {
+            expContainerName = Utils.FWD_SLASH + expContainerName.trim().toLowerCase(
+                    Locale.ENGLISH) + Utils.FWD_SLASH;
+        } else if (Utils.isNullOrEmpty(expContainerName) && !manageArtifacts) {
+            expContainerName = "default" + expContainerName.trim().toLowerCase(
+                    Locale.ENGLISH) + Utils.FWD_SLASH;
+        } else {
 			expContainerName = expContainerName.trim().toLowerCase(
-					Locale.ENGLISH);
+					Locale.ENGLISH) + Utils.FWD_SLASH;
 		}
 
 		// Resolve file path
-		String expFP = Utils.replaceTokens(build, listener, filesPath);
+		String expFP = Util.replaceMacro(filesPath, envVars);
 
 		if (expFP != null) {
 			expFP = expFP.trim();
 		}
 		
 		// Resolve exclude paths
-		String excludeFP = Utils.replaceTokens(build, listener, excludeFilesPath);
+		String excludeFP = Util.replaceMacro(excludeFilesPath, envVars);
 
 		if (excludeFP != null) {
 			excludeFP = excludeFP.trim();
 		}
 
 		// Resolve virtual path
-		String expVP = Utils.replaceTokens(build, listener, virtualPath);
-		if (Utils.isNullOrEmpty(expVP)) {
+		String expVP = Util.replaceMacro(virtualPath, envVars);
+        if (Utils.isNullOrEmpty(expVP) && !manageArtifacts) {
 			expVP = null;
 		}
 		if (!Utils.isNullOrEmpty(expVP) && !expVP.endsWith(Utils.FWD_SLASH)) {
 			expVP = expVP.trim() + Utils.FWD_SLASH;
 		}
-
-		// Validate input data
-		if (!validateData(build, listener, strAcc, expContainerName)) {
-			return true; // returning true so that build can continue.
+		
+		if (!Utils.isNullOrEmpty(expVP) && manageArtifacts) {
```

Comment:
It might be worth prepending the job folder names here.  The job/build number won't be garaunteed to be unique if folders are used in Jenkins